hadoop-hdds/rocks-native/src/test/java/org/apache/hadoop/hdds/utils/db/managed/TestManagedSSTDumpIterator.java

```diff
@@ -0,0 +1,127 @@
+    try (ManagedEnvOptions envOptions = new ManagedEnvOptions();
+         ManagedOptions managedOptions = new ManagedOptions();
+         ManagedSstFileWriter sstFileWriter = new ManagedSstFileWriter(
+             envOptions, managedOptions)) {
+      sstFileWriter.open(file.getAbsolutePath());
+      for (Map.Entry<Pair<String, Integer>, String> entry : keys.entrySet()) {
+        if (entry.getKey().getValue() == 0) {
+          sstFileWriter.delete(entry.getKey().getKey()
+              .getBytes(StandardCharsets.UTF_8));
+        } else {
+          sstFileWriter.put(entry.getKey().getKey()
+                  .getBytes(StandardCharsets.UTF_8),
+              entry.getValue().getBytes(StandardCharsets.UTF_8));
+        }
+      }
+      sstFileWriter.finish();
+      sstFileWriter.close();
+      ExecutorService executorService =
+          new ThreadPoolExecutor(1, 1, 0, TimeUnit.SECONDS,
+              new ArrayBlockingQueue<>(1),
+              new ThreadPoolExecutor.CallerRunsPolicy());
+      ManagedSSTDumpTool tool = new ManagedSSTDumpTool(executorService, 8192);
+      try (ManagedSSTDumpIterator<ManagedSSTDumpIterator.KeyValue> iterator =
+              new ManagedSSTDumpIterator<ManagedSSTDumpIterator.KeyValue>(
+              tool, file.getAbsolutePath(), new ManagedOptions()) {
+
+            @Override
+            protected KeyValue getTransformedValue(Optional<KeyValue> value) {
+              return value.orElse(null);
+            }
+          }) {
+        while (iterator.hasNext()) {
+          ManagedSSTDumpIterator.KeyValue r = iterator.next();
+          Pair<String, Integer> recordKey = Pair.of(new String(r.getKey(),
+              StandardCharsets.UTF_8), r.getType());
+          Assertions.assertTrue(keys.containsKey(recordKey));
+          Assertions.assertEquals(
+              Optional.ofNullable(keys.get(recordKey)).orElse(""),
+              new String(r.getValue(), StandardCharsets.UTF_8));
+          keys.remove(recordKey);
+        }
+        Assertions.assertEquals(0, keys.size());
+      }
+      executorService.shutdown();
+    }
+  }
+
+  @Native("Managed Rocks Tools")
+  @ParameterizedTest
+  @ArgumentsSource(KeyValueFormatArgumentProvider.class)
+  public void testSSTDumpIteratorWithKeyFormat(String keyFormat,
+              String valueFormat) throws Exception {
+    TreeMap<Pair<String, Integer>, String> keys =
+        IntStream.range(0, 100).boxed().collect(
+            Collectors.toMap(
+                i -> Pair.of(String.format(keyFormat, i), i % 2),
+                i -> i % 2 == 0 ? "" : String.format(valueFormat, i),
+                (v1, v2) -> v2,
+                TreeMap::new));
+    testSSTDumpIteratorWithKeys(keys);
+  }
+}
+
+class KeyValueFormatArgumentProvider implements ArgumentsProvider {
+  @Override
+  public Stream<? extends Arguments> provideArguments(
+      ExtensionContext context) {
+    return Stream.of(
+        Arguments.of("'key%1$d=>", "%1$dvalue'"),
```

Comment:
nit: You can also add description of the test similar to [this](https://github.com/apache/ozone/blob/master/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestLDBCli.java#L119) or [this](https://github.com/apache/ozone/blob/3858cd102bbc3c328c24261d929c6549f5358a40/hadoop-hdds/rocksdb-checkpoint-differ/src/test/java/org/apache/ozone/rocksdiff/TestRocksDBCheckpointDiffer.java#L195) so that reader will know what you are testing from the description.

Reply:
done